controllers/sandbox_controller.go

```diff
@@ -142,31 +214,29 @@ func (r *SandboxReconciler) reconcileService(ctx context.Context, sandbox *sandb
 	}
 	service.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Service"))
 	if err := ctrl.SetControllerReference(sandbox, service, r.Scheme); err != nil {
-		return nil, err
+		log.Error(err, "Failed to set controller reference")
+		return nil, fmt.Errorf("SetControllerReference for Service failed: %w", err)
 	}
 
-	if err := r.Create(ctx, service, client.FieldOwner("sandbox-controller")); err != nil {
+	err := r.Create(ctx, service, client.FieldOwner("sandbox-controller"))
+	if err != nil {
 		log.Error(err, "Failed to create", "Service.Namespace", service.Namespace, "Service.Name", service.Name)
 		return nil, err
 	}
+	sandbox.Status.ServiceFQDN = service.Name + "." + service.Namespace + ".svc.cluster.local"
```

Comment:
We still need to handle this better, given that it's not always `cluster.local`, which is the default cluster domain but customizable. Would you add a TODO and document a warning/limitation in API doc?

Reply:
done